packages/backend/src/open_payments/payment/incoming/routes.ts

```diff
@@ -51,29 +56,26 @@ async function getIncomingPayment(
   const acceptJSON = ctx.accepts('application/json')
   ctx.assert(acceptJSON, 406, 'must accept json')
 
-  const incomingPayment = await deps.incomingPaymentService.get(
-    incomingPaymentId
-  )
+  let incomingPayment: IncomingPayment | undefined
+  try {
+    incomingPayment = await deps.incomingPaymentService.get(incomingPaymentId)
+  } catch (err) {
+    ctx.throw(500, err.message)
```

Comment:
I don't know if we want to expose the error message to the client.
I vote either no message or generic message like the query resolvers:
https://github.com/interledger/rafiki/blob/51128d2ac1bf63214a8a67a19b4ec8aeea1e93b8/packages/backend/src/graphql/resolvers/outgoing_payment.ts#L79